core/src/main/scala/ai/h2o/sparkling/backend/utils/H2OContextExtensions.scala

```diff
@@ -115,21 +115,34 @@ trait H2OContextExtensions extends RestCommunication with RestApiUtils with Shel
   }
 
   protected def getAndVerifyWorkerNodes(conf: H2OConf): Array[NodeDesc] = {
+    val h2oCluster = conf.h2oCluster.get + conf.contextPath.getOrElse("")
+    val h2oClusterName = conf.cloudName.get
     try {
-      lockCloud(conf)
+      var numberOfTries = 6
+      while (numberOfTries > 1) {
+        try {
+          logInfo(s"Trying to lock H2O cluster $h2oCluster - $h2oClusterName.")
+          lockCloud(conf)
+          numberOfTries = 0
+        } catch {
+          case cause: RestApiException if numberOfTries > 1 =>
```

Comment:
interesting construct, so this means this catch only works when the `if` condition is sattisfied?

Reply:
Yes, you are correct.